lib/Driver/DarwinToolChains.cpp

```diff
@@ -262,7 +269,22 @@ toolchains::Darwin::constructInvocation(const LinkJobAction &job,
   assert(Triple.isOSDarwin());
 
   // FIXME: If we used Clang as a linker instead of going straight to ld,
-  // we wouldn't have to replicate Clang's logic here.
+  // we wouldn't have to replicate a bunch of Clang's logic here.
+
+  // Always link the regular compiler_rt if it's present.
+  //
+  // Note: Normally we'd just add this unconditionally, but it's valid to build
+  // Swift and use it as a linker without building compiler_rt.
+  SmallString<128> CompilerRTPath;
+  getClangLibraryPath(context.Args, CompilerRTPath);
+  llvm::sys::path::append(
+      CompilerRTPath,
+      Twine("libclang_rt.") +
+        getDarwinLibraryNameSuffixForTriple(Triple, /*simulator*/false) +
+        ".a");
+  if (llvm::sys::fs::exists(CompilerRTPath))
+    Arguments.push_back(context.Args.MakeArgString(CompilerRTPath));
+
```

Comment:
The routine this is in, `constructInvocation` is already too long for me to understand without breaking it down. At the least, it would be great to factor this paragraph out into a separate function. At best, in order to ease the way for others to work on this code, it would help a lot to break this `constructInvocation` function down into smaller pieces.

Reply:
Ahh, yes, I understand about cherry-picking. Sigh. Will check the other box.